### PR TITLE
pkg-config is here!

### PR DESCRIPTION
**Pull request by adamdickmeiss**

### CODE REVIEW

**Review by MikeTaylor — COMMENTED** [review]
My concern here is that we're enabling builds on systems with `pkg-config`, but at the cost of breaking them on systems without it but which do have `yaz-config`. How confident are you that all modern systems have `pkg-config`?

**Comment by adamdickmeiss** [comment]
yaz-server.pc appeared in YAZ 5.29.0. That package can be installed from our APT in pretty much all versions of Debian. For the official package, bullseye (current stable) has it. However latest Ubuntu 20.04 LTS is 5.28.0. Later versions of Ubuntu .. eg 21.04, 21.10 has ok. I'll change the script to check for the precence of yaz-server.

**Comment by MikeTaylor** [comment]
I'm also concerned about non-Debain-based systems! My MacBook with Homebrew does have `pkg-config`, which is encouraging, but there are a lot of different operating systems out there ... Windows included! So I think the way to go, if we're doing this, is definitely to go with whichever of `pkc-config` and `yaz-config` is in place.

**Comment by adamdickmeiss** [comment]
Don't get me started on Windows - there yaz-config is not around either.

pkg-config has been around pretty much forever on Debian, CentOS, RedHat.

Note: simple server requires YAZ 5.27.0 anyway.

Note with zoom-perl it's a different story. yaz.pc has been around since 2007 in YAZ (3 series) with a version that's incompatible with zoom-perl anyway!

File: Makefile.PL [hunk 1]
```diff
@@ -2,17 +2,28 @@ use ExtUtils::MakeMaker;
...
-my $yazlibs = `$yazconf --libs server`;
-if (!$yazver || (!$yazinc && !$yazlibs)) {
-   die qq[
+my $yazver;
+my $yazinc;
+my $yazlibs;
+system("pkg-config --exists yaz-server");
+if ($? == 0) {
+    $yazconf = "pkg-config";
+    $yazver = `pkg-config --modversion yaz-server` or die;
```
Comment (MikeTaylor) [inline Makefile.PL:10]: If you're going to the trouble of defining `$yazconf` you should probably use it here and in the next two lines. On the other hand, why not get rid of it completely, both here and in the alternative clause? It doesn't really buy us anything.

File: Makefile.PL [hunk 2]
```diff
@@ -2,17 +2,28 @@ use ExtUtils::MakeMaker;
...
+    $yazinc = `pkg-config --cflags yaz-server` or die;
+    $yazlibs = `pkg-config --libs yaz-server` or die;
+} else {
+    my $yazconf = "yaz-config";
+    $yazver = `$yazconf --version`;
+    $yazinc = `$yazconf --cflags servers`;
+    $yazlibs = `$yazconf --libs server`;
+    if (!$yazver || (!$yazinc && !$yazlibs)) {
+       die qq[
 ERROR: Unable to call script: yaz-config
```
Comment (MikeTaylor) [inline Makefile.PL:19]: "... or pkg-config"

**Review by MikeTaylor — CHANGES_REQUESTED** [review]
Sorry to be picky.

File: Makefile.PL (hunk 2 above)
Comment (adamdickmeiss) [reply Makefile.PL:19]: Probably not. pkg-config will complain and die.. pkg-config is much more verbose than yaz-config.

File: Makefile.PL (hunk 1 above)
Comment (adamdickmeiss) [reply Makefile.PL:10]: Sure.

**Review by MikeTaylor — APPROVED** [review]
OK, let's do it.

**Comment by adamdickmeiss** [comment]
And https://github.com/indexdata/zoom-perl/pull/5 ?

**Comment by MikeTaylor** [comment]
Oh, somehow I never got notified of that one. Same changes as here, I think!

**Pull request merged**